app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/ZonePreviewBottomSheetDialogFragment.kt

```diff
@@ -0,0 +1,153 @@
+        const val EXTRA_ZONE_NAME = "zoneName"
+
+        fun newInstance(
+            zoneId: String,
+            onShowEventsClickListener: View.OnClickListener,
+            onItineraryClickListener: View.OnClickListener
+        ): ZonePreviewBottomSheetDialogFragment {
+            val args = Bundle()
+            args.putString(EXTRA_ZONE_ID, zoneId)
+            val f = ZonePreviewBottomSheetDialogFragment()
+            f.onShowEventsClickListener = onShowEventsClickListener
+            f.onItineraryClickListener = onItineraryClickListener
+            f.arguments = args
+            return f
+        }
+    }
+
+    private var zoneId: String? = null
+
+    private lateinit var previewUpcomingEventsTextView: TextView
+    private lateinit var seeEventsButton: Button
+    private lateinit var showItineraryButton: Button
+    private lateinit var previewUpComingEventsRecyclerView: RecyclerView
+    private lateinit var zoneNameTextView: TextView
+    private lateinit var zoneDescriptionTextView: TextView
+
+    private lateinit var onItineraryClickListener: View.OnClickListener
+    private lateinit var onShowEventsClickListener: View.OnClickListener
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val v = inflater.inflate(
+            R.layout.fragment_preview_zone_events_bottom_sheet,
+            container,
+            false
+        )
+
+        zoneId = arguments?.getString(EXTRA_ZONE_ID)
+
+        seeEventsButton = v.findViewById(R.id.zone_preview_show_events_button)
+        showItineraryButton = v.findViewById(R.id.zone_preview_show_itinerary_button)
+        previewUpcomingEventsTextView = v.findViewById(R.id.zone_preview_dialog_upcoming_events)
+        previewUpComingEventsRecyclerView =
+            v.findViewById(R.id.zone_preview_dialog_event_recycler_view)
+        zoneNameTextView = v.findViewById(R.id.zone_preview_dialog_zone_name)
+        zoneDescriptionTextView = v.findViewById(R.id.zone_preview_dialog_zone_description)
+
+        val zoneObservable = Observable<Zone>()
+        zoneObservable.observeOnce(this) {
+            val zoneName = it.value.zoneName
+
+            zoneNameTextView.text = zoneName
+            zoneDescriptionTextView.text = it.value.description
+
+            // Buttons are disabled until the zone information is retrieved
+            seeEventsButton.isEnabled = true
+            showItineraryButton.isEnabled = true
+
+            // on click go to the events list per zone fragment (and dismiss the dialog as well on click)
+            seeEventsButton.setOnClickListener {
+                dismiss()
+                onShowEventsClickListener.onClick(view)
+            }
+            // Set the itinerary button on click listener (and dismiss the dialog as well on click)
+            showItineraryButton.setOnClickListener {
+                dismiss()
```

Comment:
Not sure why you dismissed the fragment here and also in the itineraryListener

Reply:
You want the fragment dialog to dimiss when you click on buttons 'show Itinerary' or 'see events'. Otherwise you'll have the fragment dialog always displayed on the bottom if you click on one of these buttons and you can't dismiss it. (unless you click outside the dialog)